web/src/test/auto/integrated/test_utils.js

```diff
@@ -0,0 +1,331 @@
+// // KeymanWeb test suite - processing of the Karma configuration's client.args parameter.
+
+import Device from '/@keymanapp/keyman/build/engine/device-detect/lib/index.mjs';
+
+export let DEVICE_DETECT_FAILURE = false;
+
+// If we've set things up to support Device dection without loading KMW...
+try {
+  let device = new Device();
+  device.detect();
+} catch (err) {
+  // Sets a warning flag that unit-test files can use to disable themselves.
+  DEVICE_DETECT_FAILURE = true;
+}
+
+// // Keyman test suite utility methods
+
+// var setupKMW = function(kmwOptions, timeout) {
```

Comment:
A lot of commented code here!

Reply:
Yeah.  Most of that is because it's addressing currently-unmodularized code.